### PR TITLE
fix(desktop): refresh task documents after workflow writes

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- fix the stale task-document root cause by adding explicit fresh-read paths in the Tauri host adapter for externally persisted spec, plan, and QA documents
- centralize authoritative document refreshes in the desktop query layer and make Agent Studio trigger one authoritative refresh after completed workflow document writes
- add regression coverage for stale metadata refreshes, task-detail refresh propagation, and Agent Studio document updates, plus document the refresh ownership

## Verification
- bun run typecheck
- bun run lint
- bun run test

## Summary by CodeRabbit

* **Refactor**
  * Unified document loading and refresh flow across spec/plan/qa for consistent behavior.
  * Simplified event handling to trigger single, deduplicated document reloads after workflow events.

* **New Features**
  * Host client and cache now support explicit "force-fresh" reads to surface external document updates reliably.

* **Tests**
  * Added and updated tests covering fresh-read behavior, deduplication, and cache-refresh edge cases.

* **Documentation**
  * Clarified ownership and cache/refresh strategy for task-document reads.

### CODE REVIEW

File: apps/desktop/src/state/operations/tasks/use-task-operations.test.tsx [hunk 1]
```diff
@@ -713,25 +713,39 @@ describe("use-task-operations", () => {
...
+        if (section === "plan") {
+          return {
+            markdown: currentPlanMarkdown,
+            updatedAt: currentPlanUpdatedAt,
+          };
+        }
+
+        return { markdown: "", updatedAt: null };
+      },
+    );
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/state/operations/tasks/use-task-operations.test.tsx]: Fixed in dad69928: I extracted the duplicated task-document mock implementation into a shared reader helper in the test.

File: apps/desktop/src/pages/agents/use-agent-studio-documents.ts [hunk 2]
```diff
@@ -167,37 +165,8 @@ export function useAgentStudioDocuments({
...
-        documentReloadAttemptsRef.current.delete(eventKey);
-        continue;
-      }
-
-      const triggered = reloadDocument(target.section);
-      if (triggered) {
-        documentReloadAttemptsRef.current.set(eventKey, attempts + 1);
-      }
+      reloadDocument(target.section);
+      processedDocumentToolEventsRef.current.add(eventKey);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/agents/use-agent-studio-documents.ts]: Fixed in dad69928: the hook now waits for repo context and only marks the event processed when reload scheduling succeeds.

File: apps/desktop/src/pages/agents/use-agent-studio-documents.ts [hunk 3]
```diff
@@ -167,37 +165,8 @@ export function useAgentStudioDocuments({
...
     }
   }, [activeSession, applyDocumentUpdate, planDoc, qaDoc, reloadDocument, specDoc, taskId]);
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/agents/use-agent-studio-documents.ts]: Fixed in dad69928: I added the  guard, kept the event unprocessed until repo hydration completes, and only add it to the dedupe set when  returns true.

File: apps/desktop/src/state/operations/tasks/use-spec-operations.test.tsx [hunk 4]
```diff
@@ -17,6 +17,44 @@ reactActEnvironment.IS_REACT_ACT_ENVIRONMENT = true;
...
+
+  return {
+    taskDocumentGet: mock(async (repoPath: string, taskId: string, section: string) =>
+      resolveSection(repoPath, taskId, section),
+    ),
+    taskDocumentGetFresh: mock(async (repoPath: string, taskId: string, section: string) =>
+      resolveSection(repoPath, taskId, section),
+    ),
+  };
+};
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/state/operations/tasks/use-spec-operations.test.tsx]: Fixed in dad69928: the test helper now returns fresh empty fixtures and fails fast on unexpected section names instead of silently treating them as QA.

File: apps/desktop/src/pages/agents/use-agent-studio-documents.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/agents/use-agent-studio-documents.ts]: Clarification: the follow-up fix added the `activeRepo` guard, leaves the event unprocessed until repo hydration completes, and only adds it to the dedupe set when `reloadDocument(...)` returns `true`.

File: apps/desktop/src/pages/agents/use-agent-studio-documents.test.tsx [hunk 5]
```diff
@@ -357,70 +358,83 @@ describe("useAgentStudioDocuments", () => {
...
+    try {
+      await harness.mount();
+      expect(reloadDocumentMock).toHaveBeenCalledTimes(1);
+      expect(reloadDocumentMock).toHaveBeenCalledWith("plan");
 
-      const activeSession = createAgentSessionFixture({
+      await harness.update(baseArgs);
+
+      expect(reloadDocumentMock).toHaveBeenCalledTimes(1);
+    } finally {
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/agents/use-agent-studio-documents.test.tsx:401]: Fixed in b3b7c73d: the dedupe regression test now rerenders with a fresh `activeSession` object carrying the same logical message, so the effect reruns and the assertion actually exercises event-key deduplication.
